### PR TITLE
Fixes disconnected status on kolibri library

### LINKED ISSUE #10721: Disconnected status always shown when browsing the Kolibri [Studio] library
## Observed behavior

A disconnected icon/status is displayed when a user browses the studio library.
Screenshot 2023-05-18 at 22 58 37

## Expected behavior

The disconnected icon/status should only be displayed when the studio library is unavailable/goes offline in the process of browsing its resources

## User-facing consequences

The status is misleading to the user at that any point in time.

## Steps to reproduce

- Remember to run Zerotier. This should already be set up from earlier sessions, but you may need to reopen the application. Incase you need a refresher on setup, please visit [ZeroTier](https://www.notion.so/ZeroTier-3e508bf09b7e4c72a1713b8374a43cac)
- To view the Kolibri Library when exploring libraries, set the environment variable below before you run the app file
   - `KOLIBRI_CENTRAL_CONTENT_BASE_URL=https://unstable.studio.learningequality.org`
- Go to `/learn/#/library`
- Click on the `Explore` card under the Kolibri library section

## Context

`0.16.0-alpha13`

### PR DESCRIPTION
**Pull request by akolson**
## Summary

This pr fixes the device connection status bug when exploring the Kolibri Library. It also write tests for the `ExploreLibrariesPage`.

## References

Fixes https://github.com/learningequality/kolibri/issues/10721

## Reviewer guidance

- Remember to run Zerotier. This should already be set up from earlier sessions, but you may need to reopen the application. Incase you need a refresher on setup, please visit [ZeroTier](https://www.notion.so/ZeroTier-3e508bf09b7e4c72a1713b8374a43cac)
- To view the Kolibri Library when exploring libraries, set the environment variable below before you run the app file
   - `KOLIBRI_CENTRAL_CONTENT_BASE_URL=https://unstable.studio.learningequality.org`
- Go to `/learn/#/library`
- Click on the `Explore` card under the Kolibri library section
- Also, browse any devices available and observe the behavior of the device connection status

----

## Testing checklist

- [x] Contributor has fully tested the PR manually
- [x] If there are any front-end changes, before/after screenshots are included
- [ ] Critical user journeys are covered by Gherkin stories
- [ ] Critical and brittle code paths are covered by unit tests

## PR process

- [x] PR has the correct target branch and milestone
- [x] PR has 'needs review' or 'work-in-progress' label
- [x] If PR is ready for review, a reviewer has been added. (Don't use 'Assignees')
- [ ] If this is an important user-facing change, PR or related issue has a 'changelog' label
- [ ] If this includes an internal dependency change, a link to the diff is provided

## Reviewer checklist

- Automated test coverage is satisfactory
- PR is fully functional
- PR has been tested for [accessibility regressions](http://kolibri-dev.readthedocs.io/en/develop/manual_testing.html#accessibility-a11y-testing)
- External dependency files were updated if necessary (`yarn` and `pip`)
- Documentation is updated
- Contributor is in AUTHORS.md

### CODE REVIEW

**Review by jredrejo — COMMENTED** [review]
Hi @akolson looking at your changes I'm not sure that's the way to go. In the future the same problem will appear with any device not been detected by zeroconf (i.e. anyone that passes through the router). Maybe the quickest solution would be not using this status icon in devices not available in zeroconf. Or if we decide to use it I think another procedure should be used, maybe one based on ping or other strategies more reliable when a router is in the middle of the path.
@bjester I'd love to know your opinion on this as the current code uses the composables you created to detect connectivity.

**Comment by akolson** [comment]
@jredrejo I hear you on this. Not sure but @rtibbles can weigh in as well. It looks like the two endpoints (devices and kolibri studio work in isolation thus the need to merge the result from the two into one. My motivation to do it that way was based off the implementation of the parallel `useDevices` composable in the learn plugin where a similar principal is applied to cater for kolibri studio library that is not returned by `fetchDevices`. This is certainly hacky and I think we need to refine further.

File: kolibri/core/assets/src/views/sync/SelectDeviceModalGroup/SelectDeviceForm.vue [hunk 1]
```diff
@@ -159,7 +169,7 @@
         useDevicesResult = useDevicesWithChannel(props.filterByChannelId);
       } else if (props.filterByFacilityId !== null) {
         // This is inherently filtered to full-facility devices
-        useDevicesResult = useDevicesWithFacility(props.filterByFacilityId);
+        useDevicesResult = useDevicesWithFacility({ facilityId: props.filterByFacilityId });
```
Comment (rtibbles) [inline kolibri/core/assets/src/views/sync/SelectDeviceModalGroup/SelectDeviceForm.vue:172]: I guess this also opens the way for filtering by devices with facilities with specific characteristics/facility settings too?

**Review by rtibbles — COMMENTED** [review]
I think this makes sense, and while @jredrejo's concerns are valid, we would have to wait until https://github.com/learningequality/kolibri/issues/10431 to implement this in a more seamless way.

File: kolibri/core/assets/src/views/sync/SelectDeviceModalGroup/SelectDeviceForm.vue (hunk 1 above)
Comment (akolson) [reply kolibri/core/assets/src/views/sync/SelectDeviceModalGroup/SelectDeviceForm.vue:172]: I think so

**Comment by radinamatic** [comment]
Tested the asset on 2 different VMs, in one as an admin on the group learning setup, and on the other as 'on-my-own' user setup. In both cases the disconnected status from the Kolibri Studio was not extant anymore.


https://github.com/learningequality/kolibri/assets/1457929/1f2577b9-1e85-410b-8f0a-0017784cf700

The only thing that would be good to clarify is the exact name of the environment variable, as when I initially set `KOLIBRI_CENTRAL_CONTENT_BASE_URL=https://unstable.studio.learningequality.org` I was unable to see the Kolibri Library in both VMs. 

Only after I changed it to `CENTRAL_CONTENT_BASE_URL=https://unstable.studio.learningequality.org`, that is without the initial `KOLIBRI_` (see the recording above), it did appear in the Library tab.

**Review by radinamatic — APPROVED** [review]
Manual QA passed, disconnected status is no more, good to go! 👏🏽 💯 :shipit:

**Comment by akolson** [comment]
@rtibbles, Radina's comment on environment variable?

**Pull request merged**